cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/Transform.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.api;
+
+/**
+ * Transform Stage
+ * @param <I> input
+ * @param <O> output
+ */
+public abstract class Transform<I, O> implements StageLifecycle {
+
+  private TransformContext context;
+
+  /**
+   * Configure the Transform stage.
```

Comment:
we should add more javadocs describing when this is called.  For somebody extending this class, the difference between configure and initialize should be clear.